File: order/order.go
package order

import (
	"fmt"
	"time"
)

type OrderService struct {
	roundRepository *RoundRepository
	orderRepository *OrderRepository
}

type Round struct {
	Round int
}

type Order struct {
	Id      int
	OrderNo string
	Round   int

	GuessNumber  int
	ResultNumber int

	BetAmount int
	WinAmount int
	Odds      int
	Status    string

	CreatedTime string
	UpdatedTime string
}

func (api OrderService) GenerateOrder(guessNumber int, betAmount int) *Order {
	currentRound := api.roundRepository.GetNextRound()

	orderNo := fmt.Sprintf("ORD-%d-%d", currentRound, time.Now().UnixNano())
	now := time.Now().Format(time.RFC3339)

	order := Order{
		OrderNo:     orderNo,
		Round:       currentRound,
		GuessNumber: guessNumber,
		BetAmount:   betAmount,
		Status:      "PENDING",
		CreatedTime: now,
		UpdatedTime: now,
	}

	api.orderRepository.CreateOrder(order)
	return &order
}

func (api OrderService) UpdateOrder(round int, resultNumber int) {
	orders := api.orderRepository.GetRoundOrders(round)

	for _, order := range orders {
		order.ResultNumber = resultNumber
		order.UpdatedTime = time.Now().Format(time.RFC3339)

		if order.GuessNumber == resultNumber {
			order.Odds = 10
			order.WinAmount = order.BetAmount * order.Odds
			order.Status = "WIN"
		} else {
			order.Odds = 0
			order.WinAmount = 0
			order.Status = "LOSE"
		}

		api.orderRepository.UpdateOrder(order)
	}
}

File: src/main/java/com/cooba/entity/Order.java
package com.cooba.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.cooba.constant.Status;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("orders")
public class Order {
    @TableId(type = IdType.AUTO)
    private Integer id;
    private String orderNo;
    private int round;

    private int guessNumber;
    private int resultNumber;

    private int betAmount;
    private int winAmount;
    private int odds;
    private Status status;
    private String result;

    private LocalDateTime createdTime;
    private LocalDateTime updatedTime;
}
